arkouda/io.py

```diff
@@ -309,404 +160,483 @@ def get_null_indices(filenames, datasets) -> Union[pdarray, Mapping[str, pdarray
+    filenames: Union[str, List[str]],
+    datasets: Optional[Union[str, List[str]]] = None,
+    iterative: bool = False,
+    strict_types: bool = True,
+    allow_errors: bool = False,
+    calc_string_offsets: bool = False,
+) -> Union[
+    pdarray,
+    Strings,
+    arkouda.array_view.ArrayView,
+    Mapping[str, Union[pdarray, Strings, arkouda.array_view.ArrayView]],
+]:
     """
-    Save multiple named pdarrays to HDF5 files.
+    Read Arkouda objects from HDF5 file/s
 
     Parameters
     ----------
-    columns : dict or list of pdarrays
-        Collection of arrays to save
-    prefix_path : str
-        Directory and filename prefix for output files
-    names : list of str
-        Dataset names for the pdarrays
-    file_format : str
-        'HDF5' or 'Parquet'. Defaults to hdf5
-    mode : {'truncate' | 'append'}
-        By default, truncate (overwrite) the output files if they exist.
-        If 'append', attempt to create new dataset in existing files.
-    file_type : str ("single" | "distribute")
-            Default: distribute
-            Single writes the dataset to a single file
-            Distribute writes the dataset to a file per locale
+    filenames : str, List[str]
+        Filename/s to read objects from
+    datasets : Optional str, List[str]
+        datasets to read from the provided files
+    iterative : bool
+        Iterative (True) or Single (False) function call(s) to server
+    strict_types: bool
+        If True (default), require all dtypes of a given dataset to have the
+        same precision and sign. If False, allow dtypes of different
+        precision and sign across different files. For example, if one
+        file contains a uint32 dataset and another contains an int64
+        dataset with the same name, the contents of both will be read
+        into an int64 pdarray.
+    allow_errors: bool
+        Default False, if True will allow files with read errors to be skipped
+        instead of failing.  A warning will be included in the return containing
+        the total number of files skipped due to failure and up to 10 filenames.
+    calc_string_offsets: bool
+        Default False, if True this will tell the server to calculate the
+        offsets/segments array on the server versus loading them from HDF5 files.
+        In the future this option may be set to True as the default.
 
     Returns
     -------
-    None
+    For a single dataset returns an Arkouda pdarray, Arkouda Strings, or Arkouda ArrayView object
+    and for multiple datasets returns a dictionary of Arkouda pdarrays,
+    Arkouda Strings or Arkouda ArrayView.
+        Dictionary of {datasetName: pdarray or String}
 
     Raises
     ------
     ValueError
-        Raised if (1) the lengths of columns and values differ or (2) the mode
-        is not 'truncate' or 'append'
-
-    See Also
-    --------
-    save, load_all
+        Raised if all datasets are not present in all hdf5/parquet files or if one or
```

Comment:
Since this is only for hdf5 files, we should probably remove the parquet reference in the docstring here

Reply:
done